discord/ext/commands/cooldowns.py

```diff
@@ -50,7 +49,7 @@
     'MaxConcurrency',
 )
 
-T = TypeVar('T')
+T_contra = TypeVar('T_contra')
```

Comment:
Seems like you're missing contravariant=True here

Reply:
Ugh, I took it out to test something... _sigh_